tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenStartThread.cs

```diff
@@ -0,0 +1,45 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public sealed class JsonDrivenStartThread : JsonDrivenWithThread
+    {
+        public JsonDrivenStartThread(JsonDrivenTestsContext testsContext, IJsonDrivenTestRunner testRunner, Dictionary<string, object> objectMap) : base(testsContext, testRunner, objectMap)
+        {
+        }
+
+        protected override void CallMethod(CancellationToken cancellationToken)
+        {
+            _testContext.Tasks.GetOrAdd(_name, (Task)null);
```

Comment:
GetOrAdd? I assume we are ADDING a new value, not GETTING an existing value.

In part this is because ConcurrentDictionary has a weird API, but perhaps you could use TryAdd instead, perhaps even asserting that TryAdd returns true.